packages/dmn-js-drd/src/Viewer.js

```diff
@@ -224,30 +224,64 @@ Viewer.prototype._createContainer = function() {
   );
 };
 
-Viewer.prototype.open = function(definitions, done) {
+/**
+ * @typedef {Object} OpenResult
+ *
+ * @property {Array<string>} warnings - Warnings occured during opening
+ */
 
-  var err;
+/**
+  * @typedef {Error} OpenError
+  *
+  * @property {Array<string>} warnings - Warnings occured during opening
+  */
 
-  // use try/catch to not swallow synchronous exceptions
-  // that may be raised during model parsing
-  try {
+/**
+ * Open diagram element.
+ *
+ * @param  {ModdleElement} definitions
+ * @returns {Promise} Resolves with {OpenResult} when successful
+ * or rejects with {OpenError}
+ */
+Viewer.prototype.open = function(definitions) {
+  var self = this;
 
-    if (this._definitions) {
+  return new Promise((resolve, reject) => {
+    var err;
 
-      // clear existing rendered diagram
-      this.clear();
-    }
+    // use try/catch to not swallow synchronous exceptions
+    // that may be raised during model parsing
+    try {
 
-    // update definitions
-    this._definitions = definitions;
+      if (self._definitions) {
 
-    // perform graphical import
-    return importDRD(this, definitions, done);
-  } catch (e) {
-    err = e;
-  }
+        // clear existing rendered diagram
+        self.clear();
+      }
+
+      // update definitions
+      self._definitions = definitions;
 
-  return done(err);
+      // perform graphical import
+      return importDRD(self, definitions, function(err, warnings) {
+        if (err) {
+          err.warnigns = warnings || [];
```

Comment:
little typo: `err.warnings`

I wonder why this never fails, though ...

Reply:
Thanks, fixed.

I also added respective test coverage (at least improved it), but in this area it is again difficult since it is really hard to find cases where you have an error + warnings (so the error needs to be covered, so that the parsing first finds warnings, before then blowing up with an error)